### PR TITLE
Add measurement info

### PR DESCRIPTION
**Pull request by Petyo98**

### CODE REVIEW

File: src/main/java/io/helti/config/constants/DbConstants.java [hunk 1]
```diff
@@ -25,7 +25,13 @@ public interface DbConstants {
 	String TBL_USERS_FAVORITE_FOODS = "users_favorite_foods";
 	String TBL_USERS_REPORTED_FOODS = "users_reported_foods";
 
+	String TBL_FOOD_MEASUREMENT_INFO =  "food_measurement_info";
+
+	String TBL_MEASURING_UNIT =  "measuring_unit";
```
Comment (DeyvidDimitrov) [inline src/main/java/io/helti/config/constants/DbConstants.java:30]: Isn't measurement better than measuring, as it is used above? I think it's more consistent.

File: src/main/java/io/helti/config/constants/DbConstants.java [hunk 2]
```diff
@@ -47,6 +53,8 @@ public interface DbConstants {
 	String SEQ_USERS_FAVORITE_FOODS = "users_favorite_foods_id_seq";
 	String SEQ_USERS_REPORTED_FOODS = "users_reported_foods_id_seq";
 
+	String SEQ_MEASURING_UNIT_ID = "measuring_unit_id_seq";
```
Comment (DeyvidDimitrov) [inline src/main/java/io/helti/config/constants/DbConstants.java:56]: Same as above.

File: src/main/java/io/helti/converter/food/FoodConverter.java [hunk 3]
```diff
@@ -41,6 +48,9 @@ public FoodDetailsDto toFoodDetailsDto(Food food) {
 		if (food.getReportedFoods() != null) {
 			dto.setReports(food.getReportedFoods().size());
 		}
+		if (food.getFoodMeasurementInfos() != null) {
```
Comment (DeyvidDimitrov) [inline src/main/java/io/helti/converter/food/FoodConverter.java:51]: Check if its not needed to add `!food.getFoodMeasurementInfos().isEmpty()`

File: src/main/java/io/helti/dto/food/FoodMeasurementInfoDto.java [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+public class FoodMeasurementInfoDto {
```
Comment (DeyvidDimitrov) [inline src/main/java/io/helti/dto/food/FoodMeasurementInfoDto.java:13]: Can't it be Record?

File: src/main/java/io/helti/dto/food/FoodMeasurementInfoRequest.java [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+import jakarta.validation.constraints.Min;
+import jakarta.validation.constraints.NotNull;
+
+public record FoodMeasurementInfoRequest(
+        @NotNull(message = "Food id required.")
+        long foodId,
+        MeasuringUnit measuringUnit,
+        @Min(value = 0, message = "Ratio must be greater than or equal to {value}.")
+        double ratio
+
```
Comment (DeyvidDimitrov) [inline src/main/java/io/helti/dto/food/FoodMeasurementInfoRequest.java:14]: No empty line.

File: src/main/java/io/helti/entity/food/Food.java [hunk 6]
```diff
@@ -72,6 +73,9 @@ public class Food {
 	@OneToMany(mappedBy = "food", cascade = CascadeType.DETACH)
 	private List<UserReportedFood> reportedFoods;
 
+	@OneToMany(mappedBy = "foodId", cascade = CascadeType.DETACH)
```
Comment (DeyvidDimitrov) [inline src/main/java/io/helti/entity/food/Food.java:76]: Are you sure about DETACH? I'm asking, don't know if its good or bad.

File: src/main/java/io/helti/entity/food/FoodMeasurementInfo.java [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
+    @Id
+    @SequenceGenerator(
+            name = "food_measurement_info_id_seq",
+            sequenceName = "food_measurement_info_id_seq",
+            allocationSize = 1)
+    @GeneratedValue(
+            strategy = GenerationType.SEQUENCE,
+            generator = "food_measurement_info_id_seq")
+    @Column(name = "food_measurement_info_id")
```
Comment (DeyvidDimitrov) [inline src/main/java/io/helti/entity/food/FoodMeasurementInfo.java:27]: Column name of id is always 'id' across the whole application, keep the consistency.

File: src/main/java/io/helti/entity/food/MeasuringUnit.java [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+            name = DbConstants.SEQ_MEASURING_UNIT_ID,
+            sequenceName = DbConstants.SEQ_MEASURING_UNIT_ID,
+            allocationSize = DbConstants.ALLOCATION_SIZE
+    )
+    @GeneratedValue(
+            strategy = GenerationType.SEQUENCE,
+            generator = DbConstants.SEQ_MEASURING_UNIT_ID
+    )
+    @Column(name = "id", nullable = false)
+    private Long unitId;
```
Comment (DeyvidDimitrov) [inline src/main/java/io/helti/entity/food/MeasuringUnit.java:27]: Why not just 'id' like across the whole application?

File: src/main/java/io/helti/service/food/FoodMeasurementInfoService.java [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+    public FoodMeasurementInfoDto createMeasurementInfo(FoodMeasurementInfoRequest foodMeasurementInfo){
+        FoodMeasurementInfo measurementInfo = foodMeasurementInfoConverter.toFoodMeasurementInfo(foodMeasurementInfo);
+        Food foodId = measurementInfo.getFoodId();
+        if (foodId == null || measurementInfo.getFoodId().getId() < 1) {
+            throw new InvalidValueException("Invalid food id.");
+        }
+
+        boolean isExistsMeasuringInfo = foodMeasurementInfoRepository.existsMeasurement(
+                measurementInfo.getRatio(), measurementInfo.getMeasuringUnitId(), measurementInfo.getFoodId());
+        if (isExistsMeasuringInfo){
```
Comment (DeyvidDimitrov) [inline src/main/java/io/helti/service/food/FoodMeasurementInfoService.java:34]: Doesn't 'isExists' look strange to you? Check for better naming probably ask ChatGPT.

File: src/main/resources/application.properties [hunk 10]
```diff
@@ -11,6 +11,6 @@ spring.application.name=helti-backend
 server.port=8080
 spring.jackson.serialization.fail-on-empty-beans=false
 application.jwt.secretKey=securesecuresecuresecuresecuresecuresecuresecuresecuresecuresecure
-application.jwt.tokenPrefix=Bearer 
+application.jwt.tokenPrefix=Bearer
```
Comment (DeyvidDimitrov) [inline src/main/resources/application.properties:14]: Why did you removed the space?

File: src/main/java/io/helti/entity/food/MeasuringUnit.java [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+import lombok.*;
+
+@Entity
+@Table(name = DbConstants.TBL_MEASURING_UNIT, schema = DbConstants.SCHEMA_PUBLIC)
+@Builder
+@Getter
+@Setter
+@NoArgsConstructor
+@AllArgsConstructor
+public class MeasuringUnit {
```
Comment (DeyvidDimitrov) [inline src/main/java/io/helti/entity/food/MeasuringUnit.java:14]: We should introduce BaseEntity probably. Raise an Asana task to have a discussion about BaseEntity.

File: src/main/java/io/helti/config/constants/DbConstants.java (hunk 2 above)
Comment (Petyo98) [reply src/main/java/io/helti/config/constants/DbConstants.java:56]: DONE

File: src/main/java/io/helti/config/constants/DbConstants.java (hunk 1 above)
Comment (Petyo98) [reply src/main/java/io/helti/config/constants/DbConstants.java:30]: DONE

File: src/main/java/io/helti/dto/food/FoodMeasurementInfoRequest.java (hunk 5 above)
Comment (Petyo98) [reply src/main/java/io/helti/dto/food/FoodMeasurementInfoRequest.java:14]: done

File: src/main/java/io/helti/entity/food/Food.java (hunk 6 above)
Comment (Petyo98) [reply src/main/java/io/helti/entity/food/Food.java:76]: yes, with .All we will delete all information in all tables, we had bug with this.

File: src/main/java/io/helti/entity/food/MeasuringUnit.java (hunk 8 above)
Comment (Petyo98) [reply src/main/java/io/helti/entity/food/MeasuringUnit.java:27]: done

File: src/main/java/io/helti/entity/food/MeasuringUnit.java (hunk 11 above)
Comment (Petyo98) [reply src/main/java/io/helti/entity/food/MeasuringUnit.java:14]: done

File: src/main/java/io/helti/service/food/FoodMeasurementInfoService.java (hunk 9 above)
Comment (Petyo98) [reply src/main/java/io/helti/service/food/FoodMeasurementInfoService.java:34]: done

File: src/main/java/io/helti/entity/food/FoodMeasurementInfo.java (hunk 7 above)
Comment (Petyo98) [reply src/main/java/io/helti/entity/food/FoodMeasurementInfo.java:27]: done

File: src/main/resources/application.properties (hunk 10 above)
Comment (Petyo98) [reply src/main/resources/application.properties:14]: мислих, че е проблем

File: src/main/java/io/helti/converter/food/FoodConverter.java (hunk 3 above)
Comment (Petyo98) [reply src/main/java/io/helti/converter/food/FoodConverter.java:51]: is better to be this way 
with is empty result is 
    },
    "currentUserMetadata": {
        "isFavorite": false,
        "isReported": false
    },
    "measurementInfo": null,
    "reports": 0,
    "verified": true

File: src/main/java/io/helti/dto/food/FoodMeasurementInfoDto.java (hunk 4 above)
Comment (Petyo98) [reply src/main/java/io/helti/dto/food/FoodMeasurementInfoDto.java:13]: ползвам го на няколко места и създавам обекти, мисля, че по добре да е така

File: src/main/java/io/helti/entity/food/MeasuringUnit.java (hunk 8 above)
Comment (Petyo98) [reply src/main/java/io/helti/entity/food/MeasuringUnit.java:27]: тук с ид не захваща ид-то изобщо не разбрах от къде идва проблема ще го оставя така, но ще сложа един коментар

**Review by DeyvidDimitrov — APPROVED** [review]
Мърджвай си го, ако нямаш какво да добавяш към този бранч от своя страна.

**Pull request merged**